sriov_ib_cni_test.sh

```diff
@@ -142,35 +149,72 @@ function exit_code {
     exit $status
 }
 
-status=0
-echo "Creating pod test-pod-1"
-pod_create 'test-pod-1'
-let status=status+$?
+function test_pods {
+    status=0
 
-if [ "$status" != 0 ]; then
-    echo "Error: error in creating the first pod"
-    exit_code $status
-fi
+    echo "Creating pod test-pod-1"
+    pod_create 'test-pod-1'
+    let status=status+$?
+    
+    if [ "$status" != 0 ]; then
+        echo "Error: error in creating the first pod"
+        return $status
+    fi
+    
+    echo "Creating pod test-pod-2"
+    pod_create 'test-pod-2'
+    let status=status+$?
+    
+    if [ "$status" != 0 ]; then
+        echo "Error: error in creating the second pod"
+        return $status
+    fi
 
-echo "Creating pod test-pod-2"
-pod_create 'test-pod-2'
-let status=status+$?
+    echo ""
+    echo "Pods created successfully! VFs GUIDs after pods creation are:"
+    after_creation_guids=$(get_vfs_guids)
+    echo "$after_creation_guids"
+    
+    test_pods_connectivity 'test-pod-1' 'test-pod-2'
+     
+    let status=status+$?
+    
+    test_guids_reseted
 
-if [ "$status" != 0 ]; then
-    echo "Error: error in creating the second pod"
-    exit_code $status
-fi
+    let status=status+$?
 
-test_pods 'test-pod-1' 'test-pod-2'
+     if [ "$status" != 0 ]; then
+        echo "Error: error in testing the pods"
+        return $status
+     fi
 
-let status=status+$?
+     echo "all tests succeeded!!"
+
+     return $status
+    
+}
+
+status=0
 
-test_guids_reseted
+
+starting_guids=$(get_vfs_guids)
+
+echo ""
+echo "VFs GUIDs before the test are:"
+echo "$starting_guids"
```

Comment:
Is there a reason you are printing `starting_guid` twice ?

Reply:
i print all the guids two times:
    - When the guid is changed, to give more information at the same location the guid changed.
    - At the end of the test, guids for all stages are printed as a summary to ease the debugging.

Reply:
basically, i print the guids twice:
* When it first change, to have more information at the place it happened.
* At the end of the test to have a summary of all the guids during the test (easier to debug).